### DISCUSSION TITLE
Authentication (publickey) failed.

### DISCUSSION THREAD

**Discussion by pigen01** [hpcaitech/ColossalAI#1104] [Community | Q&A]
在使用分布式训练样例代码时
```
colossalai run --nproc_per_node 1 --host host1 --master_addr master resnet.py
```
报如下信息
```
INFO     colossalai - paramiko.transport - INFO: Connected (version 2.0, client OpenSSH_7.6p1)                                                                             
INFO     colossalai - paramiko.transport - INFO: Authentication (publickey) failed.                                                                                        
INFO     colossalai - paramiko.transport - INFO: Authentication (publickey) failed.                                                                                        
INFO     colossalai - paramiko.transport - INFO: Authentication (publickey) failed. 
```
这是为什么？明明两台机器已经可以无密码ssh登录了

**Comment by FrankLeeeee** [comment]
Hi, do two machines use port 22 for ssh?

**Comment by pigen01** [comment]
yes，they use port 22



---- Replied Message ----
| From | Frank ***@***.***> |
| Date | 06/16/2022 10:16 |
| To | ***@***.***> |
| Cc | ***@***.******@***.***> |
| Subject | Re: [hpcaitech/ColossalAI] Authentication (publickey) failed. (Discussion #1104) |

Hi, does two machines use port 22 for ssh?

**Reply by FrankLeeeee** [reply]
That's quite strange. There are several things you may want to check
1. Can you ssh without password from one machine to the other i.e. host1 -> host2 and host2 -> host1
2. Check if you require a specific identity file, i.e. does you ssh command have the `-i` flag or is there identity file path in your ssh config (`~/.ssh/config`)

You may want to provide your ssh command here as well.

**Reply by frankxyy** [reply]
@FrankLeeeee 请问分布式训练代码可以做定制化修改来兼容机器之间ssh的不同情况吗？ 有时候机器环境修改还是挺麻烦的

**Reply by binmakeswell** [reply]
> @FrankLeeeee 请问分布式训练代码可以做定制化修改来兼容机器之间ssh的不同情况吗？ 有时候机器环境修改还是挺麻烦的

请问“定制化修改来兼容机器之间ssh的不同情况”具体是指什么场景下，期望是什么呢？

**Reply by frankxyy** [reply]
> and here as well.

比如支持含密码、指定端口、指定登录后执行脚本
